art/estimators/object_detection/pytorch_yolo.py

```diff
@@ -397,50 +423,161 @@ def predict(self, x: np.ndarray, batch_size: int = 128, **kwargs) -> List[Dict[s
+            x_preprocessed = torch.from_numpy(x_preprocessed / norm_factor).to(self.device)
+        else:
+            x_preprocessed = torch.stack([transform(x_i / norm_factor).to(self.device) for x_i in x_preprocessed])
 
-        predictions_xcycwh = self._model(x_preprocessed_tensor)
+        results_list = []
 
-        predictions_x1y1x2y2_tensor = translate_predictions_xcycwh_to_x1y1x2y2(
-            y_pred_xcycwh=predictions_xcycwh, input_height=self.input_shape[1], input_width=self.input_shape[2]
-        )
-        predictions_x1y1x2y2_array: List[Dict[str, np.ndarray]] = []
+        # Run prediction
+        num_batch = int(np.ceil(len(x_preprocessed) / float(batch_size)))
+        for m in range(num_batch):
+            # Batch using indices
+            i_batch = x_preprocessed[m * batch_size : (m + 1) * batch_size]
 
-        for i_prediction, _ in enumerate(predictions_x1y1x2y2_tensor):
-            predictions_x1y1x2y2_array.append({})
+            with torch.no_grad():
+                predictions_xcycwh = self._model(i_batch)
 
-            predictions_x1y1x2y2_array[i_prediction]["boxes"] = (
-                predictions_x1y1x2y2_tensor[i_prediction]["boxes"].detach().cpu().numpy()
+            predictions_x1y1x2y2 = translate_predictions_xcycwh_to_x1y1x2y2(
+                y_pred_xcycwh=predictions_xcycwh, input_height=self.input_shape[1], input_width=self.input_shape[2]
             )
-            predictions_x1y1x2y2_array[i_prediction]["labels"] = (
-                predictions_x1y1x2y2_tensor[i_prediction]["labels"].detach().cpu().numpy()
-            )
-            predictions_x1y1x2y2_array[i_prediction]["scores"] = (
-                predictions_x1y1x2y2_tensor[i_prediction]["scores"].detach().cpu().numpy()
-            )
-            if "masks" in predictions_x1y1x2y2_tensor[i_prediction]:
-                predictions_x1y1x2y2_array[i_prediction]["masks"] = (
-                    predictions_x1y1x2y2_tensor[i_prediction]["masks"].detach().cpu().numpy().squeeze()
-                )
 
-        return predictions_x1y1x2y2_array
+            for prediction_x1y1x2y2 in predictions_x1y1x2y2:
+                prediction = {}
 
-    def fit(self, x: np.ndarray, y, batch_size: int = 128, nb_epochs: int = 20, **kwargs) -> None:
-        raise NotImplementedError
+                prediction["boxes"] = prediction_x1y1x2y2["boxes"].detach().cpu().numpy()
+                prediction["labels"] = prediction_x1y1x2y2["labels"].detach().cpu().numpy()
+                prediction["scores"] = prediction_x1y1x2y2["scores"].detach().cpu().numpy()
+                if "masks" in prediction_x1y1x2y2:
+                    prediction["masks"] = prediction_x1y1x2y2["masks"].detach().cpu().numpy().squeeze()
+
+                results_list.append(prediction)
+
+        # Apply postprocessing
+        predictions = self._apply_postprocessing(preds=results_list, fit=False)
+
+        return predictions  # type: ignore
```

Comment:
Yes, this is necessary since since `self._apply_postprocessing` is type annotated to return `np.ndarray`. The method does not necessarily always return that type, but this is what mypy believes which causes the typing error.